lib/mappings/mappings.js

```diff
@@ -23,11 +23,10 @@ const amendMappingsBasedOnNyplSource = (data, nyplSource) => {
 
 const makeMappingsGetter = (type, nyplSource) => {
   let mappings = require(`./${type}-mapping.json`)
-  // If nyplSource given, trim mappings to agree with it:
-  if (nyplSource) mappings = amendMappingsBasedOnNyplSource(mappings, nyplSource)
-
-  return (propertyName, bib) => {
-    return mappings[propertyName]
+  return (propertyName, record) => {
+    // If nyplSource given, trim mappings to agree with it:
+    if (record.nyplSource) mappings = amendMappingsBasedOnNyplSource(mappings, record.nyplSource)
```

Comment:
Turns out this modifies the `mappings` lookup for subsequent calls. See [fix here](https://github.com/NYPL/research-catalog-indexer/pull/23)